src/providers/activeLearning/objectDetection.ts

```diff
@@ -0,0 +1,195 @@
+import axios from "axios";
+import * as tf from "@tensorflow/tfjs";
+import { ElectronProxyHandler } from "./electronProxyHandler";
+import { LocalFileSystemProxy, ILocalFileSystemProxyOptions } from "../../providers/storage/localFileSystemProxy";
+
+// tslint:disable-next-line:interface-name
+export interface DetectedObject {
+    bbox: [number, number, number, number];  // [x, y, width, height]
+    class: string;
+    score: number;
+}
+
+export class ObjectDetection {
+    private modelLoaded: boolean = false;
+
+    get loaded(): boolean {
+        return this.modelLoaded;
+    }
+
+    private model: tf.GraphModel;
+    private jsonClasses: JSON;
+
+    /**
+     * Dispose the tensors allocated by the model. You should call this when you
+     * are done with the model.
+     */
+    public dispose() {
+        if (this.model) {
+            this.model.dispose();
+        }
+    }
+
+    public async load(modelFolderPath: string) {
```

Comment:
Instead of having your `ObjectDetection` object handle the different types of loading a model - why don't we implement our own `IModelLoader` or `IModelProvider` which can be implemented to load a model as needed.  Then the `ObjectDetection` class can be simplified and just accept a `IModelProvider` during construction.

Reply:
Not sure I see a real advantage here.  Let's talk about it.